File: src/platform/tauri/commands/pty.ts
/**
 * PTY Commands - Tauri Platform Wrapper
 *
 * Wrappers for Tauri invoke commands related to pseudoterminal operations
 */

import { invoke } from '../invoke';

export const ptyCommands = {
  spawn: (options: {
    id: string;
    command: string;
    args: string[];
    cols: number;
    rows: number;
    cwd: string;
  }): Promise<void> =>
    invoke<void>('spawn_pty', options),

  write: (id: string, data: number[]): Promise<void> =>
    invoke<void>('write_to_pty', { id, data }),

  resize: (id: string, cols: number, rows: number): Promise<void> =>
    invoke<void>('resize_pty', { id, cols, rows }),

  kill: (id: string): Promise<void> =>
    invoke<void>('kill_pty', { id }),
};

File: src-tauri/src/commands/pty.rs
use tauri::State;
use crate::pty::PtyManager;

#[tauri::command]
pub async fn spawn_pty(
    pty_manager: State<'_, PtyManager>,
    id: String,
    command: String,
    args: Vec<String>,
    cols: u16,
    rows: u16,
    cwd: Option<String>,
) -> Result<String, String> {
    pty_manager
        .spawn(id, command, args, cols, rows, cwd)
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn resize_pty(
    pty_manager: State<'_, PtyManager>,
    id: String,
    cols: u16,
    rows: u16,
) -> Result<(), String> {
    pty_manager
        .resize(&id, cols, rows)
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn write_to_pty(
    pty_manager: State<'_, PtyManager>,
    id: String,
    data: Vec<u8>,
) -> Result<(), String> {
    pty_manager
        .write(&id, data)
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn kill_pty(
    pty_manager: State<'_, PtyManager>,
    id: String,
) -> Result<(), String> {
    pty_manager
        .kill(&id)
        .map_err(|e| e.to_string())
}
